rsa.go

```diff
@@ -73,7 +73,7 @@ func (m *SigningMethodRSA) Sign(signingString string, key interface{}) ([]byte,
 
 	// Validate type of key
 	if rsaKey, ok = key.(*rsa.PrivateKey); !ok {
-		return nil, ErrInvalidKey
+		return nil, newError("rsa sign expects *rsa.PrivateKey", ErrInvalidKeyType)
```

Comment:
Not sure about this one, I wasn't quite sure what the reason is that this on is a ErrInvalidKey and not a ErrInvalidKeyType, but maybe there was is a reason.

Reply:
Yes, this should be ErrInvalidKeyType. I think the original idea was to differentiate between the wrong (go) type (ErrInvalidKeyType) and invalid parameters on the key itself (ErrInvalidKey), e.g., if the curve bits did not match, etc.